### DISCUSSION TITLE
How to realize left mouse click to pause and double click to full screen without pauses

### DISCUSSION THREAD

**Discussion by 351492384fate** [mpvnet-player/mpv.net#560] [Q&A]
Like Youtube, you can click to pause and double click to full screen.
But I'm  wanted to do the same on mpv, so I added these into input.conf.

`MBTN_LEFT       cycle pause
MBTN_LEFT_DBL   cycle fullscreen`

it does work, but it also pauses when double click, and I don't want that.
I'd like to ask if there's any solution

**Comment by zhongfly** [comment]
https://github.com/natural-harmonia-gropius/input-event
try this script

example https://github.com/natural-harmonia-gropius/input-event#click-to-pause-double-click-to-fullscreen

**Comment by Jayden-Agri** [comment]
Did you find another solution? The script doesn't fit me as it changes other behaviour

**Comment by vaner-org** [comment]
I register the following actions in input.conf

```
MBTN_LEFT   script-message single-click
MBTN_LEFT_DBL  script-message double-click
```

then I use the script below, by saving it as a .lua file in mpv.net > scripts

```
local timer = nil
local timer2 = nil
local SINGLE_CLICK_MS = 60 --milliseconds
local DOUBLE_CLICK_MS = 120 --milliseconds

mp.register_script_message("single-click", function()
    if timer then
        timer:kill()
        timer = nil
    end
    if timer2 then
        timer2:kill()
        timer2 = nil
    end
    timer = mp.add_timeout(SINGLE_CLICK_MS / 1000, function()
        timer = nil
        mp.command("script-message-to mpvnet play-pause")
        timer2 = mp.add_timeout(DOUBLE_CLICK_MS / 1000, function()
            timer2 = nil
        end)
    end)
end)

mp.register_script_message("double-click", function()
    if timer then
        timer:kill()
        timer = nil
    end
    if timer2 then
        timer2:kill()
        timer2 = nil
        mp.command("script-message-to mpvnet play-pause")
    end
    mp.command("cycle fullscreen")
end)
```
